tests/unit/test_package_module_apt_get

```diff
@@ -0,0 +1,175 @@
+# Will cause the package module to call these instead.
+os.environ['CFENGINE_TEST_DPKG_CMD'] = cwd + "/mock_dpkg"
+os.environ['CFENGINE_TEST_DPKG_DEB_CMD'] = cwd + "/mock_dpkg_deb"
+os.environ['CFENGINE_TEST_DPKG_QUERY_CMD'] = cwd + "/mock_dpkg_query"
+os.environ['CFENGINE_TEST_APT_GET_CMD'] = cwd + "/mock_apt_get"
+mock_log = cwd + "/mock_output.log"
+os.environ['CFENGINE_TEST_MOCK_LOG'] = mock_log
+try:
+    os.unlink(mock_log)
+except OSError:
+    pass
+
+# operation       = file-install/repo-install/etc.
+# input           = List of lines to give process on stdin.
+# no_lines        = The number of lines that the output should have.
+# expected_output = List of chunks that should appear somewhere in output.
+# mock_lines      = The number of lines that mock_output should have.
+# mock_output     = List of chunks that should appear somewhere in the mock log.
+#                   The mock log is the command line(s) that the mock command
+#                   was called with.
+def check(operation, input, no_lines, expected_output, mock_lines, mock_output):
+    process = subprocess.Popen([apt_get_module, operation], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    for line in input:
+        process.stdin.write(line + "\n")
+    process.stdin.close()
+
+    output = ""
+    line_count = 0
+    for line in process.stdout:
+        output += line
+        line_count += 1
+
+    for chunk in expected_output:
+        if output.find(chunk) < 0:
+            sys.stdout.write("Fail: Does not appear in output: \"" + chunk + "\"\n")
+            sys.stdout.write("Actual output:\n\"" + output + "\"\n")
+            return False
+
+    if (line_count != no_lines):
+        sys.stdout.write("Fail: Incorrect number of lines returned: " + str(line_count)
+                         + " != " + str(no_lines) + " (expected)\n")
+        sys.stdout.write("Actual output:\n\"" + output + "\"\n")
+        return False
+
+    output = ""
+    line_count = 0
+    try:
+        log = open(mock_log, "r")
+        for line in log:
+            output += line
+            line_count += 1
+        log.close()
+    except IOError:
+        pass
+
+    for chunk in mock_output:
+        if output.find(chunk) < 0:
+            sys.stdout.write("Fail: Does not appear in mock log: \"" + chunk + "\"\n")
+            sys.stdout.write("Actual output:\n\"" + output + "\"\n")
+            return False
+
+    if (line_count != mock_lines):
+        sys.stdout.write("Fail: Incorrect number of lines in mock log: " + str(line_count)
+                         + " != " + str(mock_lines) + " (expected)\n")
+        sys.stdout.write("Actual output:\n\"" + output + "\"\n")
+        return False
+
+    try:
+        os.unlink(mock_log)
+    except OSError:
+        pass
+
+    return True
+
+apt_get_options = "-o Dpkg::Options::=--force-confold -o Dpkg::Options::=--force-confdef -y --force-yes"
+
+assert(check("supports-api-version", [], 1, ["1"], 0, []))
```

Comment:
Note that assert isn't a function in python; it's a keyword.  So your parentheses are spurious; and you have the option of providing a subsequent message, output on failure, characterising what just failed, e.g.
```
assert sanity_check(), 'we went crazy'
```

Fixed, along with a few other petty details, in my PR to your PR.